src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -1005,13 +1015,26 @@ public Long countObservationUnits(final Integer datasetId) {
 	}
 
 	@Override
-	public void updatePlotDatasetProperties(final Integer studyId, final List<Integer> variableIds, final String programUUID) {
+	public void updatePlotDatasetProperties(final Integer studyId, final PlotDatasetPropertiesDTO plotDatasetPropertiesDTO, final String programUUID) {
 		this.studyValidator.validate(studyId, true);
-		variableIds.forEach(this.termValidator::validate);
-		this.studyValidator.validateUpdateStudyEntryColumnsWithSupportedVariableTypes(variableIds, programUUID);
-		this.studyValidator.validateMaxStudyEntryColumnsAllowed(variableIds, programUUID);
+		plotDatasetPropertiesDTO.getVariableIds().forEach(this.termValidator::validate);
+		plotDatasetPropertiesDTO.getNameTypeIds().forEach(this.germplasmNameTypeValidator::validate);
```

Comment:
@cuenyad for these case, I think we can optimize it querying only once. I assume that it works fine for termValidator because we are caching variables.